upup/pkg/kutil/rollingupdate_cluster_validate_drain.go

```diff
@@ -0,0 +1,319 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kutil
+
+// TODO move this business logic into a service than can be called via the api
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/golang/glog"
+	api "k8s.io/kops/pkg/apis/kops"
+	validate "k8s.io/kops/pkg/validation"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
+	k8s_clientset "k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+)
+
+// RollingUpdateCluster restarts cluster nodes
+type RollingUpdateClusterDV struct {
+	Cloud fi.Cloud
+
+	MasterInterval  time.Duration
+	NodeInterval    time.Duration
+	BastionInterval time.Duration
+	K8sClient       *k8s_clientset.Clientset
+
+	ForceDrain     bool
+	FailOnValidate bool
+
+	Force bool
+
+	CloudOnly   bool
+	ClusterName string
+}
+
+// RollingUpdateData is used to pass information to perform a rolling update
+type RollingUpdateDataDV struct {
+	Cloud             fi.Cloud
+	Force             bool
+	Interval          time.Duration
+	InstanceGroupList *api.InstanceGroupList
+	IsBastion         bool
+
+	K8sClient *k8s_clientset.Clientset
+
+	ForceDrain     bool
+	FailOnValidate bool
+
+	CloudOnly   bool
+	ClusterName string
+}
+
+// TODO move retries to RollingUpdateCluster
+const retries = 8
+
+// TODO: should we check to see if api updates exist in the cluster
+// TODO: for instance should we check if Petsets exist when upgrading 1.4.x -> 1.5.x
```

Comment:
@sstarcher we need to switch to the go client as well, to get this functionality.  I am using the kubernetes api, and we need to switch to the go client.